packages/dd-trace/test/llmobs/plugins/ai/index.spec.js

```diff
@@ -53,33 +46,41 @@ describe('Plugin', () => {
     })
 
     it('creates a span for generateText', async () => {
-      await ai.generateText({
+      const options = {
         model: openai('gpt-4o-mini'),
         system: 'You are a helpful assistant',
         prompt: 'Hello, OpenAI!',
-        maxTokens: 100,
         temperature: 0.5
-      })
+      }
+
+      if (semifies(realVersion, '>=5.0.0')) {
+        options.maxOutputTokens = 100
+      } else {
+        options.maxTokens = 100
+      }
+
+      await ai.generateText(options)
 
       const { apmSpans, llmobsSpans } = await getEvents()
 
-      const expectedWorkflowSpan = expectedLLMObsNonLLMSpanEvent({
+      const expectedWorkflowMetadata = {}
+      if (semifies(realVersion, '>=5.0.0')) {
+        expectedWorkflowMetadata.maxRetries = MOCK_NUMBER
+        expectedWorkflowMetadata.maxOutputTokens = 100
+      } else {
+        expectedWorkflowMetadata.maxSteps = MOCK_NUMBER
+      }
+
+      assertLlmObsSpanEvent(llmobsSpans[0], {
         span: apmSpans[0],
         name: 'generateText',
         spanKind: 'workflow',
         inputValue: 'Hello, OpenAI!',
         outputValue: MOCK_STRING,
-        metadata: {
-          maxTokens: 100,
-          temperature: 0.5,
-          maxSteps: MOCK_NUMBER,
-          maxRetries: MOCK_NUMBER,
-        },
-        tokenMetrics: { input_tokens: MOCK_NUMBER, output_tokens: MOCK_NUMBER, total_tokens: MOCK_NUMBER },
-        tags: { ml_app: 'test', language: 'javascript', integration: 'ai' },
+        metadata: expectedWorkflowMetadata,
```

Comment:
This now tests for less properties. That is probably expected and before these were ignored?

Reply:
The extra properties were not being asserted before, I think because we were iterating over the actual keys: https://github.com/DataDog/dd-trace-js/blob/3e68da87721831c8c837d90ecfd6246c96441ef4/packages/dd-trace/test/llmobs/util.js#L16, which would mean extra keys in the expected were never checked. the new function fixes this, however.